tests/OperationProcessor.spec.ts

```diff
@@ -233,11 +250,273 @@ describe('OperationProcessor', async () => {
+  });
+
+  it('should ignore update operation with an invalid key id', async () => {
+    await operationProcessor.processBatch([createOp!]);
+
+    const updatePayload = {
+      didUniqueSuffix,
+      operationNumber: 1,
+      previousOperationHash: createOp!.getOperationHash(),
+      patch: [{
+        op: 'replace',
+        path: '/publicKey/1',
+        value: {
+          id: '#key2',
+          type: 'RsaVerificationKey2018',
+          owner: 'did:sidetree:updateid1',
+          publicKeyPem: process.hrtime() // Some dummy value that's not used.
+        }
+      }]
+    };
+
+    // Generate operation with an invalid key
+    const updateOperationBuffer = await OperationGenerator.generateUpdateOperationBuffer(updatePayload, '#InvalidKeyId', privateKey);
+    const updateOp = await addBatchFileOfOneOperationToCas(updateOperationBuffer, cas, 1, 1, 0);
+    await operationProcessor.processBatch([updateOp]);
+
+    const didDocument = await operationProcessor.resolve(didUniqueSuffix);
+
+    expect(didDocument).toBeDefined();
+    const publicKey2 = Document.getPublicKey(didDocument!, 'key2');
+    expect(publicKey2).toBeDefined();
+    expect(publicKey2!.owner).toBeUndefined(); // if update above went through, the owner would be defined
+  });
+
+  it('should ignore update operation with an invalid signature', async () => {
+    await operationProcessor.processBatch([createOp!]);
+
+    const updatePayload = {
+      didUniqueSuffix,
+      operationNumber: 1,
+      previousOperationHash: createOp!.getOperationHash(),
+      patch: [{
+        op: 'replace',
+        path: '/publicKey/1',
+        value: {
+          id: '#key2',
+          type: 'RsaVerificationKey2018',
+          owner: 'did:sidetree:updateid1',
+          publicKeyPem: process.hrtime() // Some dummy value that's not used.
+        }
+      }]
+    };
+
+    // Generate operation with an invalid key
+    const updateOperation = await OperationGenerator.generateUpdateOperation(updatePayload, '#key1', privateKey);
+    updateOperation.signature = 'InvalidSignature';
+    const updateOperationBuffer = Buffer.from(JSON.stringify(updateOperation));
+    const anchoredUpdateOperation = await addBatchFileOfOneOperationToCas(updateOperationBuffer, cas, 1, 1, 0);
+    await operationProcessor.processBatch([anchoredUpdateOperation]);
+
+    const didDocument = await operationProcessor.resolve(didUniqueSuffix);
+
+    expect(didDocument).toBeDefined();
+    const publicKey2 = Document.getPublicKey(didDocument!, 'key2');
+    expect(publicKey2).toBeDefined();
+    expect(publicKey2!.owner).toBeUndefined(); // if update above went through, the owner would be defined
+  });
+
+  it('should pick earlier of two conflicting updates', async () => {
+    await operationProcessor.processBatch([createOp!]);
+
+    const update1Payload = {
+      didUniqueSuffix,
+      operationNumber: 1,
```

Comment:
Unless it is going to break tests, consider not having adding dependencies to `operationNumber` since we agreed to remove it (issue #172).

Reply:
Removed all operationNumber in the tests.